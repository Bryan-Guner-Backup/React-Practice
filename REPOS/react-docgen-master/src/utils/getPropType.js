/**
 * Copyright (c) Facebook, Inc. and its affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 *
 * @flow
 */

/*eslint no-use-before-define: 0*/

import { namedTypes as t } from 'ast-types';
import { getDocblock } from '../utils/docblock';
import getMembers from './getMembers';
import getPropertyName from './getPropertyName';
import isRequiredPropType from '../utils/isRequiredPropType';
import printValue from './printValue';
import resolveToValue from './resolveToValue';
import resolveObjectKeysToArray from './resolveObjectKeysToArray';
import resolveObjectValuesToArray from './resolveObjectValuesToArray';
import type { PropTypeDescriptor, PropDescriptor, Importer } from '../types';

function getEnumValues(path: NodePath, importer: Importer) {
  const values = [];

  path.get('elements').each(function (elementPath) {
    if (t.SpreadElement.check(elementPath.node)) {
      const value = resolveToValue(elementPath.get('argument'), importer);

      if (t.ArrayExpression.check(value.node)) {
        // if the SpreadElement resolved to an Array, add all their elements too
        return values.push(...getEnumValues(value, importer));
      } else {
        // otherwise we'll just print the SpreadElement itself
        return values.push({
          value: printValue(elementPath),
          computed: !t.Literal.check(elementPath.node),
        });
      }
    }

    // try to resolve the array element to it's value
    const value = resolveToValue(elementPath, importer);
    return values.push({
      value: printValue(value),
      computed: !t.Literal.check(value.node),
    });
  });

  return values;
}

function getPropTypeOneOf(argumentPath: NodePath, importer: Importer) {
  const type: PropTypeDescriptor = { name: 'enum' };
  let value = resolveToValue(argumentPath, importer);
  if (!t.ArrayExpression.check(value.node)) {
    value =
      resolveObjectKeysToArray(value, importer) ||
      resolveObjectValuesToArray(value, importer);
    if (value) {
      type.value = getEnumValues(value, importer);
    } else {
      // could not easily resolve to an Array, let's print the original value
      type.computed = true;
      type.value = printValue(argumentPath);
    }
  } else {
    type.value = getEnumValues(value, importer);
  }
  return type;
}

function getPropTypeOneOfType(argumentPath: NodePath, importer: Importer) {
  const type: PropTypeDescriptor = { name: 'union' };
  if (!t.ArrayExpression.check(argumentPath.node)) {
    type.computed = true;
    type.value = printValue(argumentPath);
  } else {
    type.value = argumentPath.get('elements').map(function (itemPath) {
      const descriptor: PropTypeDescriptor = getPropType(itemPath, importer);
      const docs = getDocblock(itemPath);
      if (docs) {
        descriptor.description = docs;
      }
      return descriptor;
    });
  }
  return type;
}

function getPropTypeArrayOf(argumentPath: NodePath, importer: Importer) {
  const type: PropTypeDescriptor = { name: 'arrayOf' };

  const docs = getDocblock(argumentPath);
  if (docs) {
    type.description = docs;
  }

  const subType = getPropType(argumentPath, importer);

  if (subType.name === 'unknown') {
    type.value = printValue(argumentPath);
    type.computed = true;
  } else {
    type.value = subType;
  }
  return type;
}

function getPropTypeObjectOf(argumentPath: NodePath, importer: Importer) {
  const type: PropTypeDescriptor = { name: 'objectOf' };

  const docs = getDocblock(argumentPath);
  if (docs) {
    type.description = docs;
  }

  const subType = getPropType(argumentPath, importer);

  if (subType.name === 'unknown') {
    type.value = printValue(argumentPath);
    type.computed = true;
  } else {
    type.value = subType;
  }
  return type;
}

/**
 * Handles shape and exact prop types
 */
function getPropTypeShapish(
  name: 'shape' | 'exact',
  argumentPath: NodePath,
  importer: Importer,
) {
  const type: PropTypeDescriptor = { name };
  if (!t.ObjectExpression.check(argumentPath.node)) {
    argumentPath = resolveToValue(argumentPath, importer);
  }

  if (t.ObjectExpression.check(argumentPath.node)) {
    const value = {};
    argumentPath.get('properties').each(function (propertyPath) {
      if (propertyPath.get('type').value === t.SpreadElement.name) {
        // It is impossible to resolve a name for a spread element
        return;
      }

      const propertyName = getPropertyName(propertyPath, importer);
      if (!propertyName) return;
      const descriptor: PropDescriptor | PropTypeDescriptor = getPropType(
        propertyPath.get('value'),
        importer,
      );
      const docs = getDocblock(propertyPath);
      if (docs) {
        descriptor.description = docs;
      }
      descriptor.required = isRequiredPropType(propertyPath.get('value'));
      value[propertyName] = descriptor;
    });
    type.value = value;
  }

  if (!type.value) {
    type.value = printValue(argumentPath);
    type.computed = true;
  }

  return type;
}

// eslint-disable-next-line no-unused-vars
function getPropTypeInstanceOf(argumentPath: NodePath, importer: Importer) {
  return {
    name: 'instanceOf',
    value: printValue(argumentPath),
  };
}

const simplePropTypes = [
  'array',
  'bool',
  'func',
  'number',
  'object',
  'string',
  'any',
  'element',
  'node',
  'symbol',
  'elementType',
];

const propTypes = new Map([
  ['oneOf', getPropTypeOneOf],
  ['oneOfType', getPropTypeOneOfType],
  ['instanceOf', getPropTypeInstanceOf],
  ['arrayOf', getPropTypeArrayOf],
  ['objectOf', getPropTypeObjectOf],
  ['shape', getPropTypeShapish.bind(null, 'shape')],
  ['exact', getPropTypeShapish.bind(null, 'exact')],
]);

/**
 * Tries to identify the prop type by inspecting the path for known
 * prop type names. This method doesn't check whether the found type is actually
 * from React.PropTypes. It simply assumes that a match has the same meaning
 * as the React.PropTypes one.
 *
 * If there is no match, "custom" is returned.
 */
export default function getPropType(
  path: NodePath,
  importer: Importer,
): PropTypeDescriptor {
  let descriptor;
  getMembers(path, true).some(member => {
    const node = member.path.node;
    let name;
    if (t.Literal.check(node)) {
      name = node.value;
    } else if (t.Identifier.check(node) && !member.computed) {
      name = node.name;
    }
    if (name) {
      if (simplePropTypes.includes(name)) {
        descriptor = { name };
        return true;
      } else if (propTypes.has(name) && member.argumentsPath) {
        // $FlowIssue
        descriptor = propTypes.get(name)(member.argumentsPath.get(0), importer);
        return true;
      }
    }
  });
  if (!descriptor) {
    const node = path.node;
    if (t.Identifier.check(node) && simplePropTypes.includes(node.name)) {
      descriptor = { name: node.name };
    } else if (
      t.CallExpression.check(node) &&
      t.Identifier.check(node.callee) &&
      propTypes.has(node.callee.name)
    ) {
      // $FlowIssue
      descriptor = propTypes.get(node.callee.name)(path.get('arguments', 0));
    } else {
      descriptor = { name: 'custom', raw: printValue(path) };
    }
  }
  // $FlowIssue
  return descriptor;
}
