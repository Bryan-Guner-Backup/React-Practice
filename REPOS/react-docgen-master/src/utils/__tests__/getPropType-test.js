/**
 * Copyright (c) Facebook, Inc. and its affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 *
 */

import {
  expression,
  statement,
  noopImporter,
  makeMockImporter,
} from '../../../tests/utils';
import getPropType from '../getPropType';

describe('getPropType', () => {
  it('detects simple prop types', () => {
    const simplePropTypes = [
      'array',
      'bool',
      'func',
      'number',
      'object',
      'string',
      'any',
      'element',
      'node',
      'symbol',
      'elementType',
    ];

    simplePropTypes.forEach(type =>
      expect(
        getPropType(expression('React.PropTypes.' + type), noopImporter),
      ).toEqual({
        name: type,
      }),
    );

    // It doesn't actually matter what the MemberExpression is
    simplePropTypes.forEach(type =>
      expect(
        getPropType(expression('Foo.' + type + '.bar'), noopImporter),
      ).toEqual({
        name: type,
      }),
    );

    // Doesn't even have to be a MemberExpression
    simplePropTypes.forEach(type =>
      expect(getPropType(expression(type), noopImporter)).toEqual({
        name: type,
      }),
    );
  });

  it('detects complex prop types', () => {
    expect(
      getPropType(expression('oneOf(["foo", "bar"])'), noopImporter),
    ).toEqual({
      name: 'enum',
      value: [
        { value: '"foo"', computed: false },
        { value: '"bar"', computed: false },
      ],
    });

    // line comments are ignored
    expect(
      getPropType(expression('oneOf(["foo", // baz\n"bar"])'), noopImporter),
    ).toEqual({
      name: 'enum',
      value: [
        { value: '"foo"', computed: false },
        { value: '"bar"', computed: false },
      ],
    });

    expect(
      getPropType(expression('oneOfType([number, bool])'), noopImporter),
    ).toEqual({
      name: 'union',
      value: [{ name: 'number' }, { name: 'bool' }],
    });

    // custom type
    expect(getPropType(expression('oneOfType([foo])'), noopImporter)).toEqual({
      name: 'union',
      value: [{ name: 'custom', raw: 'foo' }],
    });

    expect(getPropType(expression('instanceOf(Foo)'), noopImporter)).toEqual({
      name: 'instanceOf',
      value: 'Foo',
    });

    expect(getPropType(expression('arrayOf(string)'), noopImporter)).toEqual({
      name: 'arrayOf',
      value: { name: 'string' },
    });

    expect(getPropType(expression('objectOf(string)'), noopImporter)).toEqual({
      name: 'objectOf',
      value: { name: 'string' },
    });

    expect(
      getPropType(expression('shape({foo: string, bar: bool})'), noopImporter),
    ).toEqual({
      name: 'shape',
      value: {
        foo: {
          name: 'string',
          required: false,
        },
        bar: {
          name: 'bool',
          required: false,
        },
      },
    });

    expect(
      getPropType(expression('exact({foo: string, bar: bool})'), noopImporter),
    ).toEqual({
      name: 'exact',
      value: {
        foo: {
          name: 'string',
          required: false,
        },
        bar: {
          name: 'bool',
          required: false,
        },
      },
    });

    // custom
    expect(getPropType(expression('shape({foo: xyz})'), noopImporter)).toEqual({
      name: 'shape',
      value: {
        foo: {
          name: 'custom',
          raw: 'xyz',
          required: false,
        },
      },
    });

    // custom
    expect(getPropType(expression('exact({foo: xyz})'), noopImporter)).toEqual({
      name: 'exact',
      value: {
        foo: {
          name: 'custom',
          raw: 'xyz',
          required: false,
        },
      },
    });

    // computed
    expect(
      getPropType(expression('shape(Child.propTypes)'), noopImporter),
    ).toEqual({
      name: 'shape',
      value: 'Child.propTypes',
      computed: true,
    });

    // computed
    expect(
      getPropType(expression('exact(Child.propTypes)'), noopImporter),
    ).toEqual({
      name: 'exact',
      value: 'Child.propTypes',
      computed: true,
    });
  });

  describe('resolve identifier to their values', () => {
    const mockImporter = makeMockImporter({
      shape: statement(`
        export default {bar: PropTypes.string};
      `).get('declaration'),

      types: statement(`
        export default ["foo", "bar"];
      `).get('declaration'),

      foo: statement(`
        export default "foo";
      `).get('declaration'),

      bar: statement(`
        export default "bar";
      `).get('declaration'),

      obj: statement(`
        export default { FOO: "foo", BAR: "bar" };
      `).get('declaration'),

      arr: statement(`
        export default ["foo", "bar"];
      `).get('declaration'),

      keys: statement(`
        export default Object.keys(obj);
        import obj from 'obj';
      `).get('declaration'),

      values: statement(`
        export default Object.values(obj);
        import obj from 'obj';
      `).get('declaration'),
    });

    it('resolves variables to their values', () => {
      const propTypeExpression = statement(`
        PropTypes.shape(shape);
        var shape = {bar: PropTypes.string};
      `).get('expression');

      expect(getPropType(propTypeExpression, noopImporter)).toMatchSnapshot();
    });

    it('resolves imported variables to their values', () => {
      const propTypeExpression = statement(`
        PropTypes.shape(shape);
        import shape from 'shape';
      `).get('expression');

      expect(getPropType(propTypeExpression, mockImporter)).toMatchSnapshot();
    });

    it('resolves simple identifier to their initialization value', () => {
      const propTypeIdentifier = statement(`
        PropTypes.oneOf(TYPES);
        var TYPES = ["foo", "bar"];
      `).get('expression');

      expect(getPropType(propTypeIdentifier, noopImporter)).toMatchSnapshot();
    });

    it('resolves importer identifier to initialization value', () => {
      const propTypeIdentifier = statement(`
        PropTypes.oneOf(TYPES);
        import TYPES from 'types';
      `).get('expression');

      expect(getPropType(propTypeIdentifier, mockImporter)).toMatchSnapshot();
    });

    it('resolves simple identifier to their initialization value in array', () => {
      const identifierInsideArray = statement(`
        PropTypes.oneOf([FOO, BAR]);
        var FOO = "foo";
        var BAR = "bar";
      `).get('expression');

      expect(
        getPropType(identifierInsideArray, noopImporter),
      ).toMatchSnapshot();
    });

    it('resolves imported identifier to their initialization value in array', () => {
      const identifierInsideArray = statement(`
        PropTypes.oneOf([FOO, BAR]);
        import FOO from 'foo';
        import BAR from 'bar';
      `).get('expression');

      expect(
        getPropType(identifierInsideArray, mockImporter),
      ).toMatchSnapshot();
    });

    it('resolves memberExpressions', () => {
      const propTypeExpression = statement(`
        PropTypes.oneOf([TYPES.FOO, TYPES.BAR]);
        var TYPES = { FOO: "foo", BAR: "bar" };
      `).get('expression');

      expect(getPropType(propTypeExpression, noopImporter)).toMatchSnapshot();
    });

    it('resolves memberExpressions from imported objects', () => {
      const propTypeExpression = statement(`
        PropTypes.oneOf([TYPES.FOO, TYPES.BAR]);
        import TYPES from 'obj';
      `).get('expression');

      expect(getPropType(propTypeExpression, mockImporter)).toMatchSnapshot();
    });

    it('correctly resolves SpreadElements in arrays', () => {
      const propTypeExpression = statement(`
        PropTypes.oneOf([...TYPES]);
        var TYPES = ["foo", "bar"];
      `).get('expression');

      expect(getPropType(propTypeExpression, noopImporter)).toMatchSnapshot();
    });

    it('correctly resolves SpreadElements in arrays from imported values', () => {
      const propTypeExpression = statement(`
        PropTypes.oneOf([...TYPES]);
        import TYPES from 'arr';
      `).get('expression');

      expect(getPropType(propTypeExpression, mockImporter)).toMatchSnapshot();
    });

    it('correctly resolves nested SpreadElements in arrays', () => {
      const propTypeExpression = statement(`
        PropTypes.oneOf([...TYPES]);
        var TYPES = ["foo", ...TYPES2];
        var TYPES2 = ["bar"];
      `).get('expression');

      expect(getPropType(propTypeExpression, noopImporter)).toMatchSnapshot();
    });

    it('does resolve object keys values', () => {
      const propTypeExpression = statement(`
        PropTypes.oneOf(Object.keys(TYPES));
        var TYPES = { FOO: "foo", BAR: "bar" };
      `).get('expression');

      expect(getPropType(propTypeExpression, noopImporter)).toMatchSnapshot();
    });

    it('resolves values from imported Object.keys call', () => {
      const propTypeExpression = statement(`
        PropTypes.oneOf(keys);
        import keys from 'keys';
      `).get('expression');

      expect(getPropType(propTypeExpression, mockImporter)).toMatchSnapshot();
    });

    it('does resolve object values', () => {
      const propTypeExpression = statement(`
        PropTypes.oneOf(Object.values(TYPES));
        var TYPES = { FOO: "foo", BAR: "bar" };
      `).get('expression');

      expect(getPropType(propTypeExpression, noopImporter)).toMatchSnapshot();
    });

    it('resolves values from imported Object.values call', () => {
      const propTypeExpression = statement(`
        PropTypes.oneOf(values);
        import values from 'values';
      `).get('expression');

      expect(getPropType(propTypeExpression, mockImporter)).toMatchSnapshot();
    });

    it('does not resolve external values without proper importer', () => {
      const propTypeExpression = statement(`
        PropTypes.oneOf(TYPES);
        import { TYPES } from './foo';
      `).get('expression');

      expect(getPropType(propTypeExpression, noopImporter)).toMatchSnapshot();
    });
  });

  it('detects custom validation functions for function', () => {
    expect(
      getPropType(expression('(function() {})'), noopImporter),
    ).toMatchSnapshot();
  });

  it('detects custom validation functions for arrow function', () => {
    expect(getPropType(expression('() => {}'), noopImporter)).toMatchSnapshot();
  });

  it('detects descriptions on nested types in arrayOf', () => {
    expect(
      getPropType(
        expression(`arrayOf(
      /**
       * test2
       */
      string
    )`),
        noopImporter,
      ),
    ).toMatchSnapshot();
  });

  it('detects descriptions on nested types in objectOf', () => {
    expect(
      getPropType(
        expression(`objectOf(
      /**
       * test2
       */
      string
    )`),
        noopImporter,
      ),
    ).toMatchSnapshot();
  });

  it('detects descriptions on nested types in shapes', () => {
    expect(
      getPropType(
        expression(`shape({
      /**
       * test1
       */
      foo: string,
      /**
       * test2
       */
      bar: bool
    })`),
        noopImporter,
      ),
    ).toMatchSnapshot();
  });

  it('detects required notations of nested types in shapes', () => {
    expect(
      getPropType(
        expression(`shape({
      foo: string.isRequired,
      bar: bool
    })`),
        noopImporter,
      ),
    ).toMatchSnapshot();
  });

  it('detects descriptions on nested types in exacts', () => {
    expect(
      getPropType(
        expression(`exact({
      /**
       * test1
       */
      foo: string,
      /**
       * test2
       */
      bar: bool
    })`),
        noopImporter,
      ),
    ).toMatchSnapshot();
  });

  it('detects required notations of nested types in exacts', () => {
    expect(
      getPropType(
        expression(`exact({
      foo: string.isRequired,
      bar: bool
    })`),
        noopImporter,
      ),
    ).toMatchSnapshot();
  });

  it('handles computed properties', () => {
    expect(
      getPropType(
        expression(`exact({
      [foo]: string.isRequired,
      bar: bool
    })`),
        noopImporter,
      ),
    ).toMatchSnapshot();
  });

  it('ignores complex computed properties', () => {
    expect(
      getPropType(
        expression(`exact({
      [() => {}]: string.isRequired,
      bar: bool
    })`),
        noopImporter,
      ),
    ).toMatchSnapshot();
  });
});
