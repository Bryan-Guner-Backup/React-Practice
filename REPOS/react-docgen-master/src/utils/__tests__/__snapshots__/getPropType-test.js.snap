// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getPropType detects custom validation functions for arrow function 1`] = `
Object {
  "name": "custom",
  "raw": "() => {}",
}
`;

exports[`getPropType detects custom validation functions for function 1`] = `
Object {
  "name": "custom",
  "raw": "function() {}",
}
`;

exports[`getPropType detects descriptions on nested types in arrayOf 1`] = `
Object {
  "description": "test2",
  "name": "arrayOf",
  "value": Object {
    "name": "string",
  },
}
`;

exports[`getPropType detects descriptions on nested types in exacts 1`] = `
Object {
  "name": "exact",
  "value": Object {
    "bar": Object {
      "description": "test2",
      "name": "bool",
      "required": false,
    },
    "foo": Object {
      "description": "test1",
      "name": "string",
      "required": false,
    },
  },
}
`;

exports[`getPropType detects descriptions on nested types in objectOf 1`] = `
Object {
  "description": "test2",
  "name": "objectOf",
  "value": Object {
    "name": "string",
  },
}
`;

exports[`getPropType detects descriptions on nested types in shapes 1`] = `
Object {
  "name": "shape",
  "value": Object {
    "bar": Object {
      "description": "test2",
      "name": "bool",
      "required": false,
    },
    "foo": Object {
      "description": "test1",
      "name": "string",
      "required": false,
    },
  },
}
`;

exports[`getPropType detects required notations of nested types in exacts 1`] = `
Object {
  "name": "exact",
  "value": Object {
    "bar": Object {
      "name": "bool",
      "required": false,
    },
    "foo": Object {
      "name": "string",
      "required": true,
    },
  },
}
`;

exports[`getPropType detects required notations of nested types in shapes 1`] = `
Object {
  "name": "shape",
  "value": Object {
    "bar": Object {
      "name": "bool",
      "required": false,
    },
    "foo": Object {
      "name": "string",
      "required": true,
    },
  },
}
`;

exports[`getPropType handles computed properties 1`] = `
Object {
  "name": "exact",
  "value": Object {
    "@computed#foo": Object {
      "name": "string",
      "required": true,
    },
    "bar": Object {
      "name": "bool",
      "required": false,
    },
  },
}
`;

exports[`getPropType ignores complex computed properties 1`] = `
Object {
  "name": "exact",
  "value": Object {
    "bar": Object {
      "name": "bool",
      "required": false,
    },
  },
}
`;

exports[`getPropType resolve identifier to their values correctly resolves SpreadElements in arrays 1`] = `
Object {
  "name": "enum",
  "value": Array [
    Object {
      "computed": false,
      "value": "\\"foo\\"",
    },
    Object {
      "computed": false,
      "value": "\\"bar\\"",
    },
  ],
}
`;

exports[`getPropType resolve identifier to their values correctly resolves SpreadElements in arrays from imported values 1`] = `
Object {
  "name": "enum",
  "value": Array [
    Object {
      "computed": false,
      "value": "\\"foo\\"",
    },
    Object {
      "computed": false,
      "value": "\\"bar\\"",
    },
  ],
}
`;

exports[`getPropType resolve identifier to their values correctly resolves nested SpreadElements in arrays 1`] = `
Object {
  "name": "enum",
  "value": Array [
    Object {
      "computed": false,
      "value": "\\"foo\\"",
    },
    Object {
      "computed": false,
      "value": "\\"bar\\"",
    },
  ],
}
`;

exports[`getPropType resolve identifier to their values does not resolve external values without proper importer 1`] = `
Object {
  "computed": true,
  "name": "enum",
  "value": "TYPES",
}
`;

exports[`getPropType resolve identifier to their values does resolve object keys values 1`] = `
Object {
  "name": "enum",
  "value": Array [
    Object {
      "computed": false,
      "value": "\\"FOO\\"",
    },
    Object {
      "computed": false,
      "value": "\\"BAR\\"",
    },
  ],
}
`;

exports[`getPropType resolve identifier to their values does resolve object values 1`] = `
Object {
  "name": "enum",
  "value": Array [
    Object {
      "computed": false,
      "value": "\\"bar\\"",
    },
    Object {
      "computed": false,
      "value": "\\"foo\\"",
    },
  ],
}
`;

exports[`getPropType resolve identifier to their values resolves imported identifier to their initialization value in array 1`] = `
Object {
  "name": "enum",
  "value": Array [
    Object {
      "computed": false,
      "value": "\\"foo\\"",
    },
    Object {
      "computed": false,
      "value": "\\"bar\\"",
    },
  ],
}
`;

exports[`getPropType resolve identifier to their values resolves imported variables to their values 1`] = `
Object {
  "name": "shape",
  "value": Object {
    "bar": Object {
      "name": "string",
      "required": false,
    },
  },
}
`;

exports[`getPropType resolve identifier to their values resolves importer identifier to initialization value 1`] = `
Object {
  "name": "enum",
  "value": Array [
    Object {
      "computed": false,
      "value": "\\"foo\\"",
    },
    Object {
      "computed": false,
      "value": "\\"bar\\"",
    },
  ],
}
`;

exports[`getPropType resolve identifier to their values resolves memberExpressions 1`] = `
Object {
  "name": "enum",
  "value": Array [
    Object {
      "computed": false,
      "value": "\\"foo\\"",
    },
    Object {
      "computed": false,
      "value": "\\"bar\\"",
    },
  ],
}
`;

exports[`getPropType resolve identifier to their values resolves memberExpressions from imported objects 1`] = `
Object {
  "name": "enum",
  "value": Array [
    Object {
      "computed": false,
      "value": "\\"foo\\"",
    },
    Object {
      "computed": false,
      "value": "\\"bar\\"",
    },
  ],
}
`;

exports[`getPropType resolve identifier to their values resolves simple identifier to their initialization value 1`] = `
Object {
  "name": "enum",
  "value": Array [
    Object {
      "computed": false,
      "value": "\\"foo\\"",
    },
    Object {
      "computed": false,
      "value": "\\"bar\\"",
    },
  ],
}
`;

exports[`getPropType resolve identifier to their values resolves simple identifier to their initialization value in array 1`] = `
Object {
  "name": "enum",
  "value": Array [
    Object {
      "computed": false,
      "value": "\\"foo\\"",
    },
    Object {
      "computed": false,
      "value": "\\"bar\\"",
    },
  ],
}
`;

exports[`getPropType resolve identifier to their values resolves values from imported Object.keys call 1`] = `
Object {
  "name": "enum",
  "value": Array [
    Object {
      "computed": false,
      "value": "\\"FOO\\"",
    },
    Object {
      "computed": false,
      "value": "\\"BAR\\"",
    },
  ],
}
`;

exports[`getPropType resolve identifier to their values resolves values from imported Object.values call 1`] = `
Object {
  "name": "enum",
  "value": Array [
    Object {
      "computed": false,
      "value": "\\"bar\\"",
    },
    Object {
      "computed": false,
      "value": "\\"foo\\"",
    },
  ],
}
`;

exports[`getPropType resolve identifier to their values resolves variables to their values 1`] = `
Object {
  "name": "shape",
  "value": Object {
    "bar": Object {
      "name": "string",
      "required": false,
    },
  },
}
`;
